static/src/javascripts/projects/commercial/modules/header-bidding/a9/a9.ts

```diff
@@ -46,38 +50,36 @@ const requestBids = (advert, slotFlatMap) => {
 		return requestQueue;
 	}
 
-	const adUnits = getHeaderBiddingAdSlots(advert, slotFlatMap)
-		.map((slot) => new A9AdUnit(advert, slot))
-		.filter((adUnit) => !adUnit.isEmpty());
+	const adUnits = getHeaderBiddingAdSlots(advert, slotFlatMap).map(
+		(slot) => new A9AdUnit(advert, slot),
+	);
 
 	if (adUnits.length === 0) {
 		return requestQueue;
 	}
 
-	requestQueue = requestQueue
-		.then(
-			() =>
-				new Promise((resolve) => {
-					window.apstag.fetchBids({ slots: adUnits }, () => {
-						window.googletag.cmd.push(() => {
-							window.apstag.setDisplayBids();
-							resolve();
-						});
+	requestQueue = requestQueue.then(
+		() =>
+			new Promise((resolve) => {
+				window.apstag?.fetchBids({ slots: adUnits }, () => {
+					window.googletag.cmd.push(() => {
+						window.apstag?.setDisplayBids();
+						resolve();
 					});
-				}),
-		)
-		.catch(() => {});
```

Comment:
Are we sure this never errors?

Reply:
No we're not - I've added the catch back in.